### ISSUE TITLE
Cannot run cpp examples, Too many resources requested for launch

### ISSUE THREAD

**Issue by Derongan** [apache/incubator-mxnet#15747]
## Description
Running resnet cpp sample throws exception
`[19:53:52] ../cpp-package/example/resnet.cpp:281: With Error: [19:53:52] /home/ubuntu/incubator-mxnet/src/operator/nn/batch_norm.cu:527: Check failed: err == cudaSuccess (701 vs. 0) : Name: BatchNormalizationUpdateOutput ErrStr:too many resources requested for launch
` 

## Environment info (Required)
(Building MXNet from source)

```
----------Python Info----------
('Version      :', '2.7.15+')
('Compiler     :', 'GCC 7.3.0')
('Build        :', ('default', 'Nov 27 2018 23:36:35'))
('Arch         :', ('64bit', ''))
------------Pip Info-----------
No corresponding pip install for current python.
----------MXNet Info-----------
No MXNet installed.
----------System Info----------
('Platform     :', 'Linux-4.15.0-1044-aws-x86_64-with-Ubuntu-18.04-bionic')
('system       :', 'Linux')
('node         :', 'ip-10-0-0-227')
('release      :', '4.15.0-1044-aws')
('version      :', '#46-Ubuntu SMP Thu Jul 4 13:38:28 UTC 2019')
----------Hardware Info----------
('machine      :', 'x86_64')
('processor    :', 'x86_64')
Architecture:        x86_64
CPU op-mode(s):      32-bit, 64-bit
Byte Order:          Little Endian
CPU(s):              32
On-line CPU(s) list: 0-31
Thread(s) per core:  2
Core(s) per socket:  16
Socket(s):           1
NUMA node(s):        1
Vendor ID:           GenuineIntel
CPU family:          6
Model:               79
Model name:          Intel(R) Xeon(R) CPU E5-2686 v4 @ 2.30GHz
Stepping:            1
CPU MHz:             2027.602
CPU max MHz:         3000.0000
CPU min MHz:         1200.0000
BogoMIPS:            4600.04
Hypervisor vendor:   Xen
Virtualization type: full
L1d cache:           32K
L1i cache:           32K
L2 cache:            256K
L3 cache:            46080K
NUMA node0 CPU(s):   0-31
Flags:               fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx pdpe1gb rdtscp lm constant_tsc rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq ssse3 fma cx16 pcid sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand hypervisor lahf_lm abm 3dnowprefetch cpuid_fault invpcid_single pti fsgsbase bmi1 hle avx2 smep bmi2 erms invpcid rtm rdseed adx xsaveopt
----------Network Test----------
Setting timeout: 10
Timing for MXNet: https://github.com/apache/incubator-mxnet, DNS: 0.0016 sec, LOAD: 0.5988 sec.
Timing for PYPI: https://pypi.python.org/pypi/pip, DNS: 0.0046 sec, LOAD: 0.2076 sec.
Timing for FashionMNIST: https://apache-mxnet.s3-accelerate.dualstack.amazonaws.com/gluon/dataset/fashion-mnist/train-labels-idx1-ubyte.gz, DNS: 0.0073 sec, LOAD: 0.2118 sec.
Timing for Conda: https://repo.continuum.io/pkgs/free/, DNS: 0.0057 sec, LOAD: 0.0264 sec.
Timing for Gluon Tutorial(en): http://gluon.mxnet.io, DNS: 0.0926 sec, LOAD: 0.0965 sec.
Timing for Gluon Tutorial(cn): https://zh.gluon.ai, DNS: 0.0492 sec, LOAD: 0.1286 sec.
```

## Build info (Required if built from source)

Compiler (gcc/clang/mingw/visual studio):

MXNet commit hash:
7c26810c18f299b16ebd8a0b78c0f96413c90bd8

Build config:
```
cmake -GNinja -DUSE_MKL_IF_AVAILABLE=ON -DUSE_CUDA=ON -DCMAKE_CUDA_COMPILER_LAUNCHER=ccache -DCMAKE_C_COMPILER_LAUNCHER=ccache -DCMAKE_CXX_COMPILER_LAUNCHER=ccache  -DUSE_OPENCV=1 -DUSE_MKLDNN=0 -DUSE_BLAS=mkl -DUSE_CPP_PACKAGE=1 ..
ninja
```

## Error Message:
```
ubuntu@ip-10-0-0-227:~/incubator-mxnet/build/cpp-package/example$ ./resnet 
[19:53:51] ../src/io/iter_mnist.cc:110: MNISTIter: load 60000 images, shuffle=1, shape=(32,784)
[19:53:51] ../src/io/iter_mnist.cc:110: MNISTIter: load 10000 images, shuffle=1, shape=(32,784)
[19:53:51] ../cpp-package/example/resnet.cpp:234: Epoch: 0
[19:53:52] ../cpp-package/example/resnet.cpp:281: Status: FAIL
[19:53:52] ../cpp-package/example/resnet.cpp:281: With Error: [19:53:52] /home/ubuntu/incubator-mxnet/src/operator/nn/batch_norm.cu:527: Check failed: err == cudaSuccess (701 vs. 0) : Name: BatchNormalizationUpdateOutput ErrStr:too many resources requested for launch
Stack trace:
  [bt] (0) ./resnet(dmlc::LogMessageFatal::~LogMessageFatal()+0x43) [0x5555cb19c1f3]
  [bt] (1) ./resnet(void mxnet::op::BatchNormForwardImpl<mshadow::gpu, float, float>(mshadow::Stream<mshadow::gpu>*, mxnet::OpContext const&, mxnet::op::BatchNormParam const&, std::vector<mxnet::TBlob, std::allocator<mxnet::TBlob> > const&, std::vector<mxnet::OpReqType, std::allocator<mxnet::OpReqType> > const&, std::vector<mxnet::TBlob, std::allocator<mxnet::TBlob> > const&, std::vector<mxnet::TBlob, std::allocator<mxnet::TBlob> > const&)+0x334) [0x5555cb6081c4]
  [bt] (2) ./resnet(void mxnet::op::BatchNormForward<mshadow::gpu, float, float>(mxnet::OpContext const&, mxnet::op::BatchNormParam const&, std::vector<mxnet::TBlob, std::allocator<mxnet::TBlob> > const&, std::vector<mxnet::OpReqType, std::allocator<mxnet::OpReqType> > const&, std::vector<mxnet::TBlob, std::allocator<mxnet::TBlob> > const&, std::vector<mxnet::TBlob, std::allocator<mxnet::TBlob> > const&)+0x33d) [0x5555cb6089dd]
  [bt] (3) ./resnet(void mxnet::op::BatchNormCompute<mshadow::gpu>(nnvm::NodeAttrs const&, mxnet::OpContext const&, std::vector<mxnet::TBlob, std::allocator<mxnet::TBlob> > const&, std::vector<mxnet::OpReqType, std::allocator<mxnet::OpReqType> > const&, std::vector<mxnet::TBlob, std::allocator<mxnet::TBlob> > const&)+0x9f6) [0x5555cb606526]
  [bt] (4) ./resnet(mxnet::exec::FComputeExecutor::Run(mxnet::RunContext, bool)+0x6e) [0x5555cd10112e]
  [bt] (5) ./resnet(+0x2b32caa) [0x5555cd106caa]
  [bt] (6) ./resnet(mxnet::engine::ThreadedEngine::ExecuteOprBlock(mxnet::RunContext, mxnet::engine::OprBlock*)+0x435) [0x5555cd0e9a25]
  [bt] (7) ./resnet(void mxnet::engine::ThreadedEnginePerDevice::GPUWorker<(dmlc::ConcurrentQueueType)0>(mxnet::Context, bool, mxnet::engine::ThreadedEnginePerDevice::ThreadWorkerBlock<(dmlc::ConcurrentQueueType)0>*, std::shared_ptr<dmlc::ManualEvent> const&)+0x11d) [0x5555cd0ece2d]
  [bt] (8) ./resnet(std::_Function_handler<void (std::shared_ptr<dmlc::ManualEvent>), mxnet::engine::ThreadedEnginePerDevice::PushToExecute(mxnet::engine::OprBlock*, bool)::{lambda()#4}::operator()() const::{lambda(std::shared_ptr<dmlc::ManualEvent>)#1}>::_M_invoke(std::_Any_data const&, std::shared_ptr<dmlc::ManualEvent>&&)+0x4e) [0x5555cd0ed0ee]
```

## Minimum reproducible example
resnet.cpp

## Steps to reproduce
1. Launch EC2 G3 8xlarge
2. Clone mxnet, checkout 1.5.0 or 1.4.1 (both have issues)
3. Build with cmake using command in build config section above
4. Try to run sample

## What have you tried to solve it?

1. Switch to old release (1.4.1)
2. Change gencode architecture to all

**Comment by Derongan** [comment]
Update: Building with -DCMAKE_BUILD_TYPE=RELEASE made it work. Im still wondering why it doesn't otherwise

**Comment by vrakesh** [comment]
@mxnet-label-bot add [Build, Example]

**Comment by deuscovrigus** [comment]
I had the same problem on the cmake build since 1.4.0 and I can confirm that  -DCMAKE_BUILD_TYPE=RELEASE fixes it
